fuzzylboxstats.py

```diff
@@ -232,9 +232,9 @@ def run(self, main_samples=None, sub_samples=None):
                 if cursor.node.lookup != "<ws>":
                     # Insert a leading space in situations like `return(x)`.
                     choice = " " + choice
-                    if cursor.node.symbol.name[cursor.pos:] == "":
-                        # Insert trailing space if there is none
-                        choice = choice + " "
+                if cursor.node.symbol.name[cursor.pos:] == "" and cursor.node.next_term.lookup != "<ws>":
```

Comment:
I understand the `next_term` clause but not why we need `cursor.node.symbol.name[cursor.pos:] == ""` -- any clues?

Reply:
Because we may be inside of a `<ws>` terminal in which case we don't need to add trailing whitespace.

Reply:
This checks that the cursor is at the very end of the terminal.

Reply:
Does the cursor position matter?

Reply:
To give an example, if we the cursor is like this `   |   `, then we don't need to insert a trailing whitespace. If the cursor is like this `   |` and the next token is also not whitespace, then we do need to add a trailing whitespace.

Reply:
OK, I get it.